User interface

code-kiwi opened the pull request.

code-kiwi committed:
Adds function for drawing textured rectangles

code-kiwi committed:
Adds image sync for pause - Adds pause boolean and key handler for pause

code-kiwi committed:
Adds image sync for pause - Adds pause boolean and key handler for pause

code-kiwi committed:
Merges main branch - Clears sync images function

code-kiwi committed:
Merges main branch after having added sprites

code-kiwi committed:
Adds empty files for UI

code-kiwi committed:
Files for creating and destorying uis

code-kiwi committed:
Updates main loop

code-kiwi committed:
Merges main branch

code-kiwi committed:
Adds function for drawing a UI

code-kiwi committed:
Adds labels to the ui

code-kiwi committed:
Renders UI labels

code-kiwi committed:
Starts button implementation

code-kiwi committed:
Merges main branch after animations

code-kiwi committed:
Adds buttons into assets - Enables button drawing into ui

code-kiwi committed:
Adds quit button to the pause menu - Cleans ui destroy in order to avoid segfault

code-kiwi committed:
Adds new handlers for mouse interaction

code-kiwi committed:
Adds mouse interaction and pause menu handling

code-kiwi committed:
Cleans event handlers

code-kiwi committed:
Cleans event handlers

BenjaminRap commented on bonus/srcs_bonus/event_handlers_bonus/mouse_press_handler_bonus.c:
J'ai du mal à voir à quoi sert cette fonction.

BenjaminRap commented on bonus/srcs_bonus/event_handlers_bonus/pause_handlers_bonus.c:
lorsque tu appuie sur le bouton pour release, ça ne change pas la texture de ton bouton en off

BenjaminRap commented on bonus/srcs_bonus/mlx_api_bonus/t_mlx_draw_rectangle_bonus.c:
tu utilise coords->c << 2 dans t_mlx_draw_rect_texture.
d'ailleurs, si on manque de performance sur cette partie là, on pourra optimiser l'affichage des pixels avec la même technique que pour les murs.
après, ça m'étonnerais que l'UI prennent beaucoup de ressources.

BenjaminRap commented on bonus/srcs_bonus/ui/init_ui_pause_bonus.c:
étant donné que la taille du tableau de labels et les boutons est un nombre un statique, tu pourrais les stack.
Après, c'est pas important du tout.

code-kiwi committed:
Corrections after review: draw_rect_texture uses now img->bpp_factor, pause handlers reset button texture to off

code-kiwi committed:
Init and destroy player with a texture (#30)

Draws the player

Refactors animation creation in order to adapt it to specific dimensions

Creates player animation from texture

Handles player shooting animation

Adds target to the player - Adds target asset - Adds target drawing

Slows roation

Corrects weapon animation when shooting

code-kiwi committed:
Merges main branch

code-kiwi committed:
Corrects pull error with updated branch + resize algorithm + main branch merged

code-kiwi committed:
Adds resizing option when getting a texture from file

code-kiwi committed:
Adding responsive UI

code-kiwi committed:
Cleans UI button disabling: add a function for disabling and calls it for pause ui

code-kiwi committed:
Merges main branch
